src/Tec/Plugin.php

```diff
@@ -100,19 +98,108 @@ public function register() {
 		}
 
 		// Do the settings.
-		// TODO: Remove if not using settings
 		$this->get_settings();
 
 		// Start binds.
 
+		// Remove reCAPTCHA v2
+		add_filter('tribe_community_events_captcha_plugin', '__return_null' );
 
+		$this->maybe_do_recaptcha_v3();
 
 		// End binds.
 
 		$this->container->register( Hooks::class );
 		$this->container->register( Assets::class );
 	}
 
+	/**
+	 * Replace reCAPTCHA v2 with v3 if there is a license key set.
+	 *
+	 * @return void
+	 *
+	 * @since 1.0.0
+	 */
+	function maybe_do_recaptcha_v3() {
+		$ce_options = \Tribe__Events__Community__Main::getOptions();
+		$option_key = $this->get_options_prefix() . '_site_key';
+		$recaptcha_key = $ce_options[$option_key];
+
+		if ( $recaptcha_key != '' ) {
+			// Template override for the main templates (in src/views/community).
+			add_filter( 'tribe_events_template_paths', [ $this, 'template_base_paths' ] );
+
+			// Template override for module templates (under src/views/community).
+			add_filter( 'tribe_get_template_part_path', [ $this, 'custom_templates' ], 10, 4 );
+
+			// Add scripts needed for v3.
+			add_action( 'wp_head', [ $this, 'add_recaptcha_scripts' ] );
+		}
+	}
+
+	/**
+	 * Add template override location for the main template files.
+	 * (Files in src/views/community.)
+	 *
+	 * @param array $paths The template paths.
+	 *
+	 * @return array The new template paths.
+	 *
+	 * @since 1.0.0
+	 */
+	function template_base_paths( array $paths ): array {
+		$slug = "tec-labs-" . PUE::get_slug();
```

Comment:
I'd try to centralize a place for the slug, for maintenance and clarity purposes. We currently have it in the Plugin and PUE classes (for the latter, with the `extension-` prefix). I think for the template path it would be more friendly to use the plugin slug, but that's a personal preference. PUE is oriented to the update engine.

Reply:
@bordoni just FYI. What do you think about this? In general, for the extensions?